crypt/test/simulation/TestCryptSimulationAnother2dNightly.hpp

```diff
@@ -180,13 +180,13 @@ class TestCryptSimulationAnother2dNightly : public AbstractCellBasedWithTimingsT
         //double crypt_length = num_cells_depth - 1.0;
 
         HoneycombMeshGenerator generator(num_cells_width, num_cells_depth, 0);
-        MutableMesh<2,2>* p_mesh = generator.GetMesh();
+        boost::shared_ptr<MutableMesh<2,2> > p_mesh = generator.GetMesh();
         std::vector<unsigned> location_indices = generator.GetCellLocationIndices();
 
         // Set up cells
         std::vector<CellPtr> cells;
         CryptCellsGenerator<FixedG1GenerationalCellCycleModel> cells_generator;
```

Comment:
Maybe a new ticket to get the CellsGenerators to take a `shared_ptr` instead of a raw pointer?

Reply:
Thanks @mirams. I can create a new ticket to do this as well. I've left it this way because the [core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f7-for-general-use-take-t-or-t-arguments-rather-than-smart-pointers) recommend taking non-smart pointer arguments for functions in general (unless they are going to take partial responsibility for managing the object or going to pass it forward to another function that will), and `CryptCellsGenerator::Generate()` only seems to be querying some properties of the mesh at the moment.